--- a/bzl/dependencies.bzl
+++ b/bzl/dependencies.bzl
@@ -1,4 +1,5 @@
 load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")
+load("@bazel_tools//tools/build_defs/repo:utils.bzl", "maybe")
 load("//bzl:nested_repository.bzl", "nested_repository")
 
 def local_or_nested_repository(workspace_root, name, path):
@@ -120,7 +121,8 @@
         url = "https://github.com/bazelbuild/rules_shell/releases/download/v0.3.0/rules_shell-v0.3.0.tar.gz",
     )
 
-    http_archive(
+    maybe(
+        http_archive,
         name = "build_bazel_rules_apple",
         sha256 = "70b0fb2aec1055c978109199bf58ccb5008aba8e242f3305194045c271ca3cae",
         url = "https://github.com/bazelbuild/rules_apple/releases/download/4.0.0/rules_apple.4.0.0.tar.gz",
